### DISCUSSION TITLE
Polish translation

### DISCUSSION THREAD

**Discussion by MarcinWad** [giuliopons/adadmin#104] [Q&A] [answered]
I want to buy this software. Does it have some "onefile" translation that i can translate or i have to dig inside every php file?

**Comment by giuliopons** [comment, marked as answer]
No. Don't worry! There is just a language file, a simple txt, that you have to manually translate! :-)

**Comment by tavoweb** [comment]
yes but its problem with that ... then u updating script the translation file is messed no consistency and its hard to update the translation ...

**Reply by giuliopons** [reply]
That's true. I've not a solution for that. Any suggestions?

**Reply by tavoweb** [reply]
Solution is to add new strings by versions ...

Exmple 

Version 1 
String translation ...
String translation...
Version 1.0.1
String translation...
And so on ....

**Reply by giuliopons** [reply]
You mean to add a section at the end of each language file with the new lines added or modified in the last version, in this way translations are not grouped by topic... I don't know, I have to think about it.
Thank you.

**Comment by giuliopons** [comment]
In version 4.2.7 (#111) there is now a third parameter in language files which will be filled with the version number.  

For example in this block of translations I've marked the new labels with the number 4.2.7

```
"Link","Link"
"Text copied to clipboard","Text copied to clipboard"
"Media files","Media files"
"Select file","Select file","4.2.7"
"Upload max %s files;","Upload max %s files;","4.2.7"
"File size max is %sKb;","File size max is %sKb;","4.2.7"
"Image size is %sx%s pixels;","Image size is %sx%s pixels;","4.2.7"
"File already selected. Remove it to choose another file.","File already selected. Remove it to choose another file.","4.2.7"
"Campaign","Campaign"
"No reduction","No reduction"
```

This should help the maintainers of local translation files.
